build/rpmfc.cc

```diff
@@ -1640,6 +1640,60 @@ rpmRC rpmfcApply(rpmfc fc)
     return rc;
 }
 
+static string rpmfcPrettyFType(rpmfc fc, unsigned ix)
+{
+    string ftype = fc->ftype[ix];
+    size_t len = ftype.find(' ', 10);
+    return ftype.substr(0, len);
+}
+
+static rpmRC rpmfcCheckPackageColor(rpmfc fc)
+{
+    Package pkg = fc->pkg;
+    const char *a = headerGetString(pkg->header, RPMTAG_ARCH);
+    char *nvr;
+    string msg, type, bins;
+    unsigned ix;
+    int color;
```

Comment:
There's a bunch of variables unnecessarily at the function scope: type, ix and color are only needed inside the for-loop. 

Also since this isn't using goto, nvr could be moved to the bottom to the place where its used.